code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -780,27 +781,42 @@ proc/ui_describe_reagents(atom/A)
 			var/datum/lineResult/result = drawLine(src, container, "condenser", "condenser_end", src.pixel_x + 10, src.pixel_y, container.pixel_x, container.pixel_y + get_chemical_effect_position())
 			result.lineImage.pixel_x = -src.pixel_x
 			result.lineImage.pixel_y = -src.pixel_y
-			src.UpdateOverlays(result.lineImage, "tube")
-			current_container = container
+			src.UpdateOverlays(result.lineImage, "tube\ref[container]")
+			src.connected_containers.Add(container)
 
-	proc/remove_container()
-		src.UpdateOverlays(null, "tube")
-		UnregisterSignal(current_container, COMSIG_ATTACKHAND)
-		UnregisterSignal(current_container, XSIG_OUTERMOST_MOVABLE_CHANGED)
-		UnregisterSignal(current_container, COMSIG_MOVABLE_MOVED)
-		current_container = null
+	proc/remove_container(var/obj/container)
+		src.UpdateOverlays(null, "tube\ref[container]")
+		UnregisterSignal(container, COMSIG_ATTACKHAND)
+		UnregisterSignal(container, XSIG_OUTERMOST_MOVABLE_CHANGED)
+		UnregisterSignal(container, COMSIG_MOVABLE_MOVED)
+		src.connected_containers.Remove(container)
+
+	proc/remove_all_containers()
+		for(var/obj/container in src.connected_containers)
+			remove_container(container)
 
 	proc/try_adding_reagents_to_container(reagent, amount, sdata, temp_new, donotreact, donotupdate) //called when a reaction occurs inside the condenser flagged with "chemical_reaction = TRUE"
-		if(!current_container) //if we have no beaker, dump the reagents into condenser
+		if(src.connected_containers.len <= 0) //if we have no beaker, dump the reagents into condenser
 			src.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
 		else
-			var/remaining_container_space = current_container.reagents.maximum_volume - current_container.reagents.total_volume
-			if(remaining_container_space < amount) 																			//if there's more reagent to add than the beaker can hold...
-				current_container.reagents.add_reagent(reagent, remaining_container_space, sdata, temp_new, donotreact, donotupdate) //...add what we can to the beaker...
-				src.reagents.add_reagent(reagent, amount - remaining_container_space, sdata, temp_new, donotreact, donotupdate)  //...then backflow remaining chems into the condenser
-
-			else
-				current_container.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+			add_reagents_to_containers(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+
+	proc/add_reagents_to_containers(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+		var/list/non_full_containers = list()
+		for(var/obj/container in connected_containers)
+			if(container.reagents.maximum_volume > container.reagents.total_volume) //don't bother with this if it's already full, move onto other containers
+				non_full_containers.Add(container)
+		if(non_full_containers.len <= 0)	//all full? backflow!!
```

Comment:
this could be made into a guard to be a bit cleaner, ie `return` instead of `else`